frontend/__tests__/components/login/ProviderLink.test.tsx

```diff
@@ -0,0 +1,150 @@
+const githubProps: Props = {
+  inACard: true,
+  provider: githubProvider,
+}
+const translationMock = jest.fn()
+
+jest.mock("react-i18next", () => ({
+  useTranslation: () => {
+    return {
+      t: translationMock,
+      i18n: {
+        changeLanguage: () => new Promise(() => {}),
+      },
+    }
+  },
+}))
+
+const useStateSpy = jest.spyOn(React, "useState")
+const setStateMock = jest.fn()
+// @ts-ignore
+useStateSpy.mockImplementation((init) => [init, setStateMock])
+const useLocalStorageSpy = jest.spyOn(Storage.prototype, "setItem")
+
+jest.mock("react-toastify", () => ({
+  toast: {
+    success: jest.fn(),
+    error: jest.fn(),
+  },
+}))
+
+let location: Location
+beforeEach(() => {
+  location = window.location
+  jest.spyOn(window, "location", "get").mockRestore()
+  const mockedLocation = {
+    ...location,
+  }
+  jest.spyOn(window, "location", "get").mockReturnValue(mockedLocation)
+})
+
+describe("ProviderLink tests", () => {
+  it("redirect login with success", async () => {
+    const expectedRedirectValue = "https://wiki.gnome.org"
+    const expectedReturnToValue = '""'
+
+    global.fetch = jest.fn(() =>
+      Promise.resolve({
+        ok: true,
+        json: () => Promise.resolve({ redirect: expectedRedirectValue }),
+      }),
+    ) as jest.Mock
+
+    const { getByRole } = render(
+      <ProviderLink
+        provider={githubProps.provider}
+        inACard={githubProps.inACard}
+      />,
+    )
+
+    await waitFor(() => {
+      fireEvent.click(getByRole("button"))
+    })
+
+    expect(translationMock.mock.calls[0]).toEqual([
+      "login-with-provider",
+      { provider: "GitHub" },
+    ])
+    expect(setStateMock).toHaveBeenCalledWith(true)
+    expect(useLocalStorageSpy).toHaveBeenCalledWith(
+      "returnTo",
+      expectedReturnToValue,
+    )
+    expect(window.location.href).toEqual(expectedRedirectValue)
+  })
+
+  it("redirect login with api error", async () => {
+    global.fetch = jest.fn(() => Promise.reject()) as jest.Mock
+    const { getByRole } = render(
+      <ProviderLink
+        provider={githubProps.provider}
+        inACard={githubProps.inACard}
+      />,
+    )
+
+    await waitFor(() => {
+      fireEvent.click(getByRole("button"))
+    })
+
+    expect(setStateMock.mock.calls).toEqual([[true], [false]])
+    expect(translationMock).toHaveBeenCalledWith("network-error-try-again")
+    expect(toast.error).toHaveBeenCalled()
+  })
+
+  it("redirect login with response error", async () => {
+    global.fetch = jest.fn(() =>
+      Promise.resolve({
+        ok: false,
+        json: () => Promise.resolve({}),
+      }),
+    ) as jest.Mock
+
+    const { getByRole } = render(
+      <ProviderLink
+        provider={githubProps.provider}
+        inACard={githubProps.inACard}
+      />,
+    )
+
+    await waitFor(() => {
+      fireEvent.click(getByRole("button"))
+    })
+    expect(setStateMock.mock.calls).toEqual([[true], [false]])
+    expect(toast.error).toHaveBeenCalled()
```

Comment:
These expectations would also pass on the previous test, so that's probably an anti pattern. Not sure if in the test or in the components behavior.

Reply:
I agree, I'll just leave the validation of the toast text, which is the only difference between the error scenarios